code/datums/components/lock_on_cursor.dm

```diff
@@ -0,0 +1,202 @@
+		var/atom/thing = weak_target.resolve()
+		if(thing && (get_dist(thing, mouse_tracker.given_turf) > lock_cursor_range))
+			continue
+		LAZYREMOVE(locked_weakrefs, weak_target)
+
+/// Replace our targets with new ones
+/datum/component/lock_on_cursor/proc/find_targets()
+	var/mob/owner = parent
+	if(!owner.client)
+		return
+	var/list/atom/targets = get_nearest(mouse_tracker.given_turf, target_typecache, lock_amount, lock_cursor_range)
+	if(targets == LOCKON_IGNORE_RESULT)
+		return
+	LAZYCLEARLIST(locked_weakrefs)
+	for(var/atom/target as anything in targets)
+		if(immune_weakrefs[WEAKREF(target)])
+			continue
+		LAZYOR(locked_weakrefs, WEAKREF(target))
+	refresh_visuals()
+	on_lock.Invoke(locked_weakrefs)
+
+/// Returns true if target is a valid target
+/datum/component/lock_on_cursor/proc/can_target(atom/target)
+	var/mob/mob_target = target
+	return is_type_in_typecache(target, target_typecache) && !(ismob(target) && mob_target.stat != CONSCIOUS) && !immune_weakrefs[WEAKREF(target)]
+
+/// Returns the nearest targets to the current cursor position
+/datum/component/lock_on_cursor/proc/get_nearest()
+	current_ranging_id++
+	var/this_id = current_ranging_id
+	var/list/targets = list()
+	var/turf/target_turf = mouse_tracker.given_turf
+	var/turf/center = target_turf
+	if(!length(target_typecache))
+		return
+	if(lock_cursor_range == 0)
+		return typecache_filter_list(target_turf.contents + target_turf, target_typecache)
+	var/x = 0
+	var/y = 0
+	var/cd = 0
+	while(cd <= lock_cursor_range)
+		x = center.x - cd + 1
+		y = center.y + cd
+		LOCKON_RANGING_BREAK_CHECK
+		for(x in x to center.x + cd)
+			target_turf = locate(x, y, center.z)
+			if(target_turf)
+				targets |= special_list_filter(target_turf.contents, can_target_callback)
+				if(targets.len >= lock_amount)
+					targets.Cut(lock_amount+1)
+					return targets
+		LOCKON_RANGING_BREAK_CHECK
+		y = center.y + cd - 1
+		x = center.x + cd
+		for(y in center.y - cd to y)
+			target_turf = locate(x, y, center.z)
+			if(target_turf)
+				targets |= special_list_filter(target_turf.contents, can_target_callback)
+				if(targets.len >= lock_amount)
+					targets.Cut(lock_amount+1)
+					return targets
+		LOCKON_RANGING_BREAK_CHECK
+		y = center.y - cd
+		x = center.x + cd - 1
+		for(x in center.x - cd to x)
+			target_turf = locate(x, y, center.z)
+			if(target_turf)
+				targets |= special_list_filter(target_turf.contents, can_target_callback)
+				if(targets.len >= lock_amount)
+					targets.Cut(lock_amount+1)
+					return targets
+		LOCKON_RANGING_BREAK_CHECK
+		y = center.y - cd + 1
+		x = center.x - cd
+		for(y in y to center.y + cd)
+			target_turf = locate(x, y, center.z)
+			if(target_turf)
+				targets |= special_list_filter(target_turf.contents, can_target_callback)
+				if(targets.len >= lock_amount)
+					targets.Cut(lock_amount+1)
+					return targets
+		LOCKON_RANGING_BREAK_CHECK
+		cd++
+		CHECK_TICK
+
+/atom/movable/screen/fullscreen/cursor_catcher/lock_on
```

Comment:
you don't need to remove this but you don't actually need this, the atom will be defined with just the proc

Reply:
I just think it's slightly neater to define explicitly anyway.

Reply:
if you're going to do it like that can you dmdoc the difference between this and parent